test/rockpaperscissors.js

```diff
@@ -0,0 +1,772 @@
+    const INVALID_PLAYER_MSG = "Invalid player";
+
+    const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+    const ZERO_BYTES_32 = "0x0000000000000000000000000000000000000000000000000000000000000000";
+    const DAY_24_HOUR_IN_SECS = 86400;
+    const TIMEOUT = 3500000
+
+    const getBlockTimeStamp = async(txObj) => {
+
+        const tx = await getTransaction(txObj.tx);
+        const blockData = await web3.eth.getBlock(tx.blockNumber)
+
+        return blockData.timestamp;
+    }
+
+    const getGasCost = async txObj => {
+        const tx = await getTransaction(txObj.tx);
+
+        return toBN(txObj.receipt.gasUsed).mul(toBN(tx.gasPrice));
+    };
+
+    const [contractOwner, playerOne, playerTwo, playerThree] = accounts;
+    const playerOneSecretBytes32 = await soliditySha3(playerOneSecretString);
+    const playerTwoSecretBytes32 = await soliditySha3(playerTwoSecretString);
+    const wrongSecretBytes32 = await soliditySha3(wrongSecretString);
+
+    let rcp, gameKey, playerTwoMoveHash, snapshotId;
+
+    beforeEach("Deploy and prepare", async function() {
+        rcp = await Game.new({from: contractOwner});
+        gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, ROCK);
+        playerTwoMoveHash = await rcp.createPlayerTwoMoveHash(playerTwo, gameKey, playerTwoSecretBytes32, PAPER);
+        const snapShot = await timeMachine.takeSnapshot();
+        snapshotId = snapShot['result'];
+    });
+
+    afterEach(async() => {
+        await timeMachine.revertToSnapshot(snapshotId);
+    });
+
+    describe("Player moves", async () => {
+
+          it("Player one hashes their address, secret and move", async () => {
+                gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, ROCK);
+                const expectedMoveHash =  await soliditySha3(playerOne, playerOneSecretBytes32, ROCK);
+                assert.strictEqual(expectedMoveHash, gameKey);
+            });
+
+            it("Player one creates a game", async () => {
+                const txObj = await rcp.movePlayerOne(gameKey, toBN(10), {from: playerOne, value: toBN(10)});
+
+               
+
+                await truffleAssert.eventEmitted(txObj, 'PlayerOneMoves', (ev) => {
+
+                    return  ev.player === playerOne &&
+                            ev.gameKey === gameKey &&
+                            ev.stake.toString(10) === "10" &&
+                            ev.amount.toString(10) === "10"
+                },'PlayerOneMoves event is emitted');
+
+                const game = await rcp.games(gameKey);
+
+                assert.strictEqual(game.stake.toString(10), "10");
+                assert.strictEqual(game.playerOne, playerOne);
+                assert.strictEqual(game.step.toString(10), PLAYER_ONE_MOVE);
+
+                const playerOneOwed = toBN(await rcp.balances(playerOne));
+
+                assert.strictEqual(playerOneOwed.toString(10), "0");
```

Comment:
If a string is returned already on line 100, no need to bother with the double conversion.